ckanext/saml2/plugin.py

```diff
@@ -176,6 +180,34 @@ def get_came_from(relay_state):
     return came_from.encode('utf8')
 
 
+def saml2_get_userid_by_name_id(id):
+    user_info = model.Session.query(SAML2User).\
+        filter(SAML2User.name_id == id).first()
+    return user_info.id if user_info is not None else user_info
+
+
+def saml2_get_user_info(id, return_name_id=False):
+    query = model.Session.query(SAML2User).\
+        filter(or_(SAML2User.user_name == id,
+                   SAML2User.id == id))
+    if return_name_id:
+        user_info = query.first()
+        return user_info if user_info is None else user_info.name_id
+    else:
+        return query
```

Comment:
I leave it for using query.update instead of making new query for update custom table.